sympy/sets/sets.py

```diff
@@ -799,6 +799,35 @@ def Ropen(cls, a, b):
         """Return an interval not including the right boundary."""
         return cls(a, b, False, True)
 
+    @classmethod
+    def map(cls, i, ii):
+        """
+        Return an expression that maps the first interval
+        into the second.
+
+        Examples
+        ========
+
+        >> from sympy import Interval
+        >> from sympy.abc import a, b, x
+        >> from sympy import factor
+        >> e = Interval.map(Interval(10, 12), Interval(-1, 1))
+        >> print(e.subs(x, 10))
+        0
+        >> e = Interval.map(Interval(a, b), Interval(-1, 1))
+        >> print(factor(e))
+        -(-a - b + 2*x)/(a - b)
+
+        """
+        # let i = [a, b], ii = [c, d]
+        # a < x < b
+        # 0 < x - a < b - a
+        # 0 < (x - a) / (b - a) < 1
+        # 0 < (d - c) * (x - a) / (b - a) < d - c
+        # c < c + (d - c) * (x - a) / (b - a) < d
+        a, b, c, d = i.start, i.end, ii.start, ii.end
+        return c + (d - c) * (Symbol('x') - a) / (b - a)
```

Comment:
Actually, never thought of this case ... Would it be right to let the
caller specify an argument which is the symbol to use ? Thanks
for the comment !

2015-11-16 5:44 GMT+01:00 Christopher Smith <notifications@github.com>:

> In sympy/sets/sets.py
> <https://github.com/sympy/sympy/pull/10142#discussion_r44888757>:
>
> > +        >> e = Interval.map(Interval(10, 12), Interval(-1, 1))
> > +        >> print(e.subs(x, 10))
> > +        0
> > +        >> e = Interval.map(Interval(a, b), Interval(-1, 1))
> > +        >> print(factor(e))
> > +        -(-a - b + 2*x)/(a - b)
> > +
> > +        """
> > +        # let i = [a, b], ii = [c, d]
> > +        # a < x < b
> > +        # 0 < x - a < b - a
> > +        # 0 < (x - a) / (b - a) < 1
> > +        # 0 < (d - c) * (x - a) / (b - a) < d - c
> > +        # c < c + (d - c) * (x - a) / (b - a) < d
> > +        a, b, c, d = i.start, i.end, ii.start, ii.end
> > +        return c + (d - c) * (Symbol('x') - a) / (b - a)
>
> What if the interval already uses the symbol x?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sympy/sympy/pull/10142/files#r44888757>.
>